src/webgpu/util/texture/subresource.ts

```diff
@@ -43,46 +46,33 @@ export class SubresourceRange {
 
   *each(): Generator<{ level: number; slice: number }> {
     for (let level = this.mipRange.begin; level < this.mipRange.end; ++level) {
-      for (let slice = this.sliceRange.begin; slice < this.sliceRange.end; ++slice) {
+      const end =
+        this.dimension === '3d'
+          ? this.depthOrArrayLayerCount >> level
+          : this.depthOrArrayLayerCount;
+      for (let slice = this.sliceRange.begin; slice < Math.min(end, this.sliceRange.end); ++slice) {
```

Comment:
I think this function is supposed to iterate over the subresources of a resource, but 3D textures just have one subresource per level (not per level per slice). Seems like this is returning a subresource for each slice as well? See also [1]s

> Each such texture subresource is uniquely identified by a texture, mipmap level, and (for 2d textures only) array layer, and aspect.